freud/data.py

```diff
@@ -0,0 +1,178 @@
+    defined by a lattice and a basis. It provides the basic attributes of the
+    unit cell as well as enabling the generation of systems of points
+    (optionally with some noise) from the unit cell.
+
+    Args:
+        box:
+            A box-like object (see :meth:`~freud.box.Box.from_box`) containing
+            the lattice vectors of the unit cell.
+        basis_positions ((:math:`N_{points}`, 3) :class:`numpy.ndarray`):
+            The basis of the unit cell in fractional coordinates
+            (Default value = ``[[0, 0, 0]]``).
+    """
+    def __init__(self, box, basis_positions=[[0, 0, 0]]):
+        self._box = freud.box.Box.from_box(box)
+        self._basis_positions = basis_positions
+
+    def to_system(self, num_replicas=1, scale=1, sigma_noise=0, seed=None):
+        """Generate a system from the unit cell.
+
+        The box and the positions are expanded by ``scale``, and then Gaussian
+        noise with standard deviation ``sigma_noise`` is added to the
+        positions.  All points are wrapped back into the box before being
+        returned.
+
+        Args:
+            num_replicas (:class:`tuple` or int):
+                If provided as a single number, the number of replicas in all
+                dimensions. If a tuple, the number of times replicated in each
+                dimension. Must be of the form (nx, ny, 1) for 2D boxes
+                (Default value = 1).
+            scale (float):
+                Factor by which scale the unit cell (Default value = 1).
+            sigma_noise (float):
+                The standard deviation of the normal distribution used to add
+                noise to the positions in the system (Default value = 0).
+            seed (int):
+                If provided, used to seed the random noise generation. Not used
+                unless ``sigma_noise`` > 0 (Default value = None).
+
+        Returns:
+            tuple (:class:`freud.box.Box`, :class:`np.ndarray`):
+                A system-like object (see
+                :class:`~freud.locality.NeighborQuery.from_system`.).
+        """
+        try:
+            nx, ny, nz = num_replicas
+        except TypeError:
+            nx = ny = num_replicas
+            nz = 1 if self.box.is2D else num_replicas
+
+        if self.box.is2D and nz != 1:
+            raise ValueError("The number of replicas in z must be 1 for a "
+                             "2D unit cell.")
+
+        if any([n > 1 for n in (nx, ny, nz)]):
+            pbuff = freud.locality.PeriodicBuffer()
+            pbuff.compute((self.box, self.basis_positions),
+                          buffer=(nx-1, ny-1, nz-1),
+                          images=True)
+            box = pbuff.buffer_box*scale
+            positions = np.concatenate((self.basis_positions,
+                                        pbuff.buffer_points))
+        else:
+            box = self.box*scale
+            positions = self.basis_positions.copy()
+        positions *= scale
```

Comment:
Am I missing the part where you convert from fractional to absolute coordinates? I can't find it anywhere.